### PR TITLE
- Improvements for gnip #2211

### PR DESCRIPTION
**Pull request by afabiani**

### CODE REVIEW

**Comment by pjdufour** [comment]
@afabiani, could you squash your commits?

Also in the future, could you link to the issue in the PR description.  It's not clear which issue it is related to until you scroll down.  I know it's linked in reverse, but it should link both ways.  You can just use GitHub's internal linking which automatically creates a link to #2211 from  `#2211`.

Thanks!

**Comment by afabiani** [comment]
Hello Patrik,
I should have done everything. If you find any issue please report it to me.

Thanks,
-A.

File: geonode/layers/models.py [hunk 1]
```diff
@@ -88,6 +100,18 @@ class Layer(ResourceBase):
     name = models.CharField(max_length=128)
     typename = models.CharField(max_length=128, null=True, blank=True)
 
+    is_mosaic = models.BooleanField(default=False)
+    has_time = models.BooleanField(default=False)
+    has_elevation = models.BooleanField(default=False)
+    time_regex = models.CharField(max_length=128, null=True, blank=True, choices=TIME_REGEX)
+    elevation_regex = models.CharField(max_length=128, null=True, blank=True)
+
+    is_mosaic = models.BooleanField(default=False)
```
Comment (jj0hns0n) [inline geonode/layers/models.py]: Why are these here 2x?

Comment (afabiani) [inline geonode/layers/models.py]: It may be a mistake while squashing commits. Need to check

Best Regards,
Alessio Fabiani.

==
GeoServer Professional Services from the experts!
Visit http://goo.gl/it488V for more information.
==

Ing. Alessio Fabiani
@alfa7691
Founder/Technical Lead

GeoSolutions S.A.S.
Via di Montramito 3/A
55054  Massarosa (LU)
Italy
phone: +39 0584 962313
fax:     +39 0584 1660272
mob:   +39 331 6233686

http://www.geo-solutions.it
http://twitter.com/geosolutions_it

-------------------------------------------------------

*AVVERTENZE AI SENSI DEL D.Lgs. 196/2003*

Le informazioni contenute in questo messaggio di posta elettronica e/o
nel/i file/s allegato/i sono da considerarsi strettamente riservate. Il
loro utilizzo è consentito esclusivamente al destinatario del messaggio,
per le finalità indicate nel messaggio stesso. Qualora riceviate questo
messaggio senza esserne il destinatario, Vi preghiamo cortesemente di
darcene notizia via e-mail e di procedere alla distruzione del messaggio
stesso, cancellandolo dal Vostro sistema. Conservare il messaggio stesso,
divulgarlo anche in parte, distribuirlo ad altri soggetti, copiarlo, od
utilizzarlo per finalità diverse, costituisce comportamento contrario ai
principi dettati dal D.Lgs. 196/2003.



The information in this message and/or attachments, is intended solely for
the attention and use of the named addressee(s) and may be confidential or
proprietary in nature or covered by the provisions of privacy act
(Legislative Decree June, 30 2003, no.196 - Italy's New Data Protection
Code).Any use not in accord with its purpose, any disclosure, reproduction,
copying, distribution, or either dissemination, either whole or partial, is
strictly forbidden except previous formal approval of the named
addressee(s). If you are not the intended recipient, please contact
immediately the sender by telephone, fax or e-mail and delete the
information in this message that has been received in error. The sender
does not give any warranty or accept liability as the content, accuracy or
completeness of sent messages and accepts no responsibility  for changes
made after they were sent or for other risks which arise as a result of
e-mail transmission, viruses, etc.

---------------------------------------------------------------------

**Comment by jj0hns0n** [comment]
@afabiani once this duplicated stuff is fixed, Im +1 to merge. Its a HUGE PR, so we will have to rely on you guys to help deal with any issues that come up and as already discussed, this will need to be adapted to the new importer once that is ready.

**Comment by ingenieroariel** [comment]
I am +1 after the fixes Jeff mentioned have been implemented.

**Comment by afabiani** [comment]
Duplicated stuff removed
